tc/sgx/trusted_worker_manager/enclave_untrusted/enclave_bridge/signup_singleton.h

```diff
@@ -0,0 +1,37 @@
+/* Copyright 2020 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#pragma once
+
+#include <stdlib.h>
+
+#include "tcf_error.h"
+#include "types.h"
+#include "signup.h"
+
+class SignupDataSingleton : public SignupData {
```

Comment:
Why not have the 2 methods in SignupData as non-static? Is that class meant only for inheritence?

Reply:
The two functions can be made non-static. The class SignupData was meant to inherit from different workers.

Reply:
My understanding is these functions were static because it doesn't need to access any class member variables

Reply:
Correct, these functions do not access member variables. I made them static again.